src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -1046,17 +1049,47 @@ func (c *controller) updateNodesLabels() bool {
 			continue
 		}
 		hostsWithLabels[hostname] = hostData
+		allNodeLabelsAsOneString += hostData.Host.NodeLabels
+	}
+	return hostsWithLabels, strings.Contains(allNodeLabelsAsOneString, roleLabel), nil
+}
+
+// Set labels to hosts
+// If one of the labels was role pause mcp and set this role
+// after setting all labels unpause in case it was paused
+func (c *controller) setNodesLabels() bool {
+	c.log.Infof("Setting node labels if require")
+	hostsWithLabels, pauseMCO, err := c.getAssistedHostsWithLabels()
+	if err != nil {
+		return KeepWaiting
 	}
 	if len(hostsWithLabels) == 0 {
 		c.log.Infof("No hosts with labels, skipping update node labels")
 		return ExitWaiting
 	}
 
-	if len(hostsWithLabels) > c.patchNodesLabels(log, hostsWithLabels) {
+	// if it was paused already we will skip in PauseUnpauseMachineConfigPool
+	if pauseMCO {
+		err = c.kc.PauseUnpauseMachineConfigPool(true, workerMCPName)
+		if err != nil {
+			c.log.WithError(err).Warnf("Failed to pause pool %s", workerMCPName)
+			return KeepWaiting
+		}
+	}
+
+	if len(hostsWithLabels) > c.patchNodesLabels(c.log, hostsWithLabels) {
 		return KeepWaiting
-	} else {
-		return ExitWaiting
 	}
+
+	if pauseMCO {
```

Comment:
If someone clears the node labels while the MCP is already paused, then it will remain paused. Right ?

Reply:
Clears from assisted service? I believe we don't allow such api calls in the middle of installation

Reply:
actually we can always unpause though i think this case can't really happened.